Assets/Scripts/Characters/PlayerMovement.cs

```diff
@@ -53,37 +89,87 @@ private void FixedUpdate()
             _animator.SetBool(IsJumpingID, _isJumping);
         }
 
-        private void OnDisable()
+        #region Server
+
+        [Server]
+        public override void OnStopServer()
         {
             _isJumping = false;
             _animator.SetBool(IsJumpingID, _isJumping);
             _animator.SetBool(IsRunningID, false);
         }
 
-        private void HandleMovement(float movement)
+        [Command]
+        private void CmdMovementStarted()
         {
-            _body.velocity = new Vector2(movement * speed, _body.velocity.y);
+            ServerMovementStarted?.Invoke(connectionToClient.connectionId);
+        }
 
-            if (movement > 0.01f)
+        [Command]
+        private void CmdMovementEnded()
+        {
+            ServerMovementEnded?.Invoke(connectionToClient.connectionId);
+        }
+
+        [Command]
+        private void CmdJump()
+        {
+            ServerPlayerJumped?.Invoke(connectionToClient.connectionId);
+        }
+
+        #endregion
+
+        #region Client
+
+        public override void OnStartAuthority()
+        {
+            _playerInput.actions["Player/Jump"].performed += ClientHandleJump;
+        }
+
+        public override void OnStopClient()
+        {
+            if (!isClientOnly || !hasAuthority)
             {
-                _body.transform.localScale = Vector3.one;
+                return;
             }
-            else if (movement < -0.01f)
+
+            _playerInput.actions["Player/Jump"].performed -= ClientHandleJump;
+        }
+
+        private void ClientHandleMove(float movement)
+        {
+            if (!enabled)
```

Comment:
Energy exhaustion disables the script instead of disabling the game object. I don't fully understand the difference yet but disabling the gameobject basically makes the script non-existent whereas the script can partially function if it's just disabled. Hopefully we can figure out a solution where we don't have to manually use these flags